### DISCUSSION TITLE
Feature Request: Asymmetric Tiling

### DISCUSSION THREAD

**Discussion by AugmentedRealityCat** [Coyote-A/ultimate-upscale-for-automatic1111#46] [Ideas]
Even if the source picture is tiling seamlessly, once upscaled with Ultimate-Upscale the seams becomes visible.

Currently, if the seamlessly tiling source image is upscaled with the **Asymmetric Tiling extension turned ON** the results are **worse**. It seems to be highlighting the edge instead of hiding it.

With **Asymmetric Tiling turned OFF**, the seam becomes very visible on the upscaled result anyways, but it's not as **bad**.

Since you are already using tiles to split the source image, and that you are already looking at neighbouring tiles to influence the seamless continuity between them, can you somehow take the leftmost tiles and repeat them on the right side, and do the opposite, right-to-left, for the rightmost tiles ? And then apply the same principle on the vertical Y axis (top-tiles-to-bottom + bottom-tiles-to-top) if the user has selected that option ? 

Maybe it's easier to manage if you create your own Tile-in-X and Tile-in-Y checkboxes, and not bother with connecting with the existing Asymmetrical Tiling extension ? 

One use for very high resolution images is to create immersive 360 panoramas like this one I made with your extension:

https://renderstuff.com/tools/360-panorama-web-viewer-sharing/?image=https://i.imgur.com/qfKL5hO.jpg&title=SD-to-360%20cave%20prototype

But for now I have to edit the seam manually to hide the fact it is very visible after the upscale. I do that in photoshop, and it's ok, but it would be much easier to do that directly during the image upscale process.

**Comment by Coyote-A** [comment]
I need to research it but I'm afraid it is not possible now. script do many inpaints of big image with padding. script knows about neighbouring tiles because of that. but when it is near to a border of image, there are no more tiles, so it cropped. to achieve seamless tiling from one border to opposite it should copy parts of tiles to another. I experimented with copying, but it gave visible seams, so I'm not sure if copying from one border to another will help

**Comment by Coyote-A** [comment]
you can check this strategy with this steps
1) make bigger version of your image by stacking 3x3 copies of image to sides
2) pass it through upscaler
3) crop central tile in photoshop and check for seams

**Comment by AugmentedRealityCat** [comment]
I will make the test and come back to you with the results.

**Comment by Gorotsuki** [comment]
Hello,

I have performed a lot of tests and in the end, I encounter the same issue at the moment with the latest source.

To add some confusion, I can say that my very first test seemed to not having the issue, done 1 week ago, with the latest sources.
However the more I tested, the sad turth revealed itself: it is not big, but it is there, a seam.

I tried varying options of settings, orders of operations between scaling and, ultimate-upscale, all in one (jumping from low to max hires (max supported by my card :-) ) or all operations done with very little jumps of uppering scales.
Also, I tried the sames tests with and without the Tiling settings, from img2img.

I could not reproduce anymore a success.

I am wondering if you have hope to find a fix. Would you please?
Or if you have any suggestion that would assist me in a success, I welcome it.
